test/test_linearizer.py

```diff
@@ -42,16 +42,23 @@ def test_load_dedup(self):
 
   def test_load_cache_const_bufs(self):
     # make sure const buffers are differentiated from local and mem buffers
-    a = Tensor([1,2,3,4])
-    out = a[2] + 2 + a[3] + 3 + 2 + a[0]
-    si = create_schedule([out.lazydata])[-1]
-    lin = Linearizer(si.ast)
+    ST, DT = ShapeTracker(views=(View(shape=((1,)), strides=(0, 0), offset=0, mask=None, contiguous=False),)), dtypes.int
+    VAL = LazyOp(op=BufferOps.CONST, src=(), arg=ConstBuffer(val=2, dtype=DT, st=ST))
+
+    # data1[0] + VAL
+    a = LazyOp(op=BinaryOps.ADD, src=(LazyOp(op=BufferOps.LOAD, src=(), arg=MemBuffer(idx=1, dtype=DT, st=ST)), VAL))
+    # (literal const 1) + VAL
+    b = LazyOp(op=BinaryOps.ADD, src=(LazyOp(op=BufferOps.CONST, src=(), arg=ConstBuffer(val=1, dtype=DT, st=ST)), VAL))
+
+    ast = LazyOp(op=BufferOps.STORE, src=(LazyOp(op=BinaryOps.ADD, src=(a,b)),), arg=MemBuffer(idx=0, dtype=DT, st=ST))
+    lin = Linearizer(ast)
     lin.linearize()
 
-    cache_keys = lin.load_cache.keys()
-    assert len(cache_keys) == 5
-    assert len([k for k in cache_keys if "CONST" in k]) == 2
-    assert len([k for k in cache_keys if "CONST" not in k]) == 3
+    a_bufs = [u.uop for u in lin.uops[-2].vin[0].vin]
+    b_bufs = [u.uop for u in lin.uops[-2].vin[1].vin]
+
+    assert a_bufs == [UOps.LOAD, UOps.CONST]
+    assert b_bufs == [UOps.CONST, UOps.CONST]
```

Comment:
this test would fail before the fix landed:
```
tinygrad((HEAD detached at 002bf380)) python3 -m unittest test.test_linearizer.TestLinearizer.test_load_cache_const_bufs
F
======================================================================
FAIL: test_load_cache_const_bufs (test.test_linearizer.TestLinearizer)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/qazal/code/tinygrad/tinygrad/test/test_linearizer.py", line 61, in test_load_cache_const_bufs
    assert b_bufs == [UOps.CONST, UOps.CONST]
AssertionError
```